### ISSUE TITLE
how to get if method is hooked or not without logs

### ISSUE THREAD

**Issue by Jasi2169** [rovo89/XposedBridge#118]
Hi i have a question
How can i know the method i am hooking is hooked successfully without need to go in xposed logs 

Something like we do try n catch

Can this be implemented find and hook and catch if the method is not found ?

Or any other way ?

**Comment by Jasi2169** [comment]
Sorry by mistake i clicked comment now i cant find option to edit


So here is my question
How can i know the method i am hooking is hooked successfully wihtpu need to go in xposed logs 

Any way ?

**Comment by Jasi2169** [comment]
The effect ahould be immediate once checkmark to enable xposed it shud show error of method or class not found

**Comment by rovo89** [comment]
I don't understand what you mean. An Error will be thrown indeed when the method or class couldn't be found, and you can (but don't have to) catch it. See http://api.xposed.info. Is that what you mean?

**Comment by Jasi2169** [comment]
but i need to catch it
for example in details :- 
1. okay user enabled module and rebooted 
1. now user check the checkmark for hook 
   for example

and the function for that hook is this

```
findAndHookMethod("com.example.test", lpparam.classLoader, "result",int.class, new XC_MethodHook() {
            @Override
            protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                    // anything

                }
            });
```

now what i am looking for is that when user click it and somehow the class "test" or method "result" not found it immediately gets caught like we do in try catch block
something like this
try
=> hookmethod
catch
=> method or class not found
show user sorry cant be found and immediately disable the checkmark of that hook
something like this tried to change findhook then it donot work
i know the throwable stuff can be seen in logs but the effect should be immediate rather than going to xposed logs

hope i am much clear now

**Comment by rovo89** [comment]
Sorry, but I still don't understand what you want to do.

As I said and as mentioned in the [API](http://api.xposed.info/reference/de/robv/android/xposed/XposedHelpers.html#findAndHookMethod%28java.lang.String,%20java.lang.ClassLoader,%20java.lang.String,%20java.lang.Object...%29), you can catch `NoSuchMethodError`.

But I don't understand this:
"when user click it" => click what?
"show user sorry cant be found" => show it where?
"immediately disable the checkmark of that hook" => which checkmark? does your module also provide a UI or are you talking about the checkmark in Xposed Installer that enables/disables your whole module (not individual hooks)?

**Comment by Jasi2169** [comment]
hi rovo thanks ,i guess i commented first then later found out the API methods

this is what i needed
- NoSuchMethodError
- XposedHelpers.ClassNotFoundError

PS some issue is coming is new

method can be found and hooked but i did this way

```
if (lpparam.packageName.equals("z.test")){

                XposedHelpers.findAndHookMethod("z.test.Example", lpparam.classLoader, "gettest", int.class, new XC_MethodHook() {
                    @Override
                    protected void beforeHookedMethod(MethodHookParam param) throws Throwable {

                            ishooked = true;

                        param.setResult(0);
                    }
                });

        }

if (lpparam.packageName.equals(BuildConfig.APPLICATION_ID)) {
            XposedHelpers.findAndHookMethod(BuildConfig.APPLICATION_ID + ".MainActivity", lpparam.classLoader, "isXposedEnabled", XC_MethodReplacement.returnConstant(true));

            if (ishooked == true){

                XposedHelpers.findAndHookMethod(BuildConfig.APPLICATION_ID + ".MainActivity", lpparam.classLoader, "ishooked", XC_MethodReplacement.returnConstant(true));

            }

        }
```

isxposedenabled returns true as expected but ishooked doesnot return true donot know why ? but method is hooked properly in the other application and i can see changes 

seems like my package is getting hooked first thatswhy the hooked boolean becomes true next time but my package is not hooked later when hooked boolean is true something strange

**Now i dont know how to get if its hooked or not,the trick i tried above is not working neither pref.edit() works as it is deprected**

**Comment by Jasi2169** [comment]
One more thing on initzygote 
I make ishooked boolean to false

PS the main reason is that i want to get the result is method is successfully hooked or not ,unfortunately there is no API which brings the result of method after it is hooked

If you have anyway by which we can get result and the result can be used in another activity to display to user if method hooked or not ?

**Comment by rovo89** [comment]
I think you're missing some key points in how Android and Xposed work.

All (well, most) apps are run in separate processes, and process A can't access the memory of process B. So when you set `ishooked = true` in process A, this won't be reflected in process B. Similarly, even if there was an API to check whether a certain method was hooked (which wouldn't mean that it was hooked by you...), you wouldn't be able to check whether the hook was successful in another process.

Small exception: All apps processes are forked from the Zygote process, so everything that was done before that fork (e.g. in `initZygote()`) will be visible in all app processes. But that won't help you here.

So in short, you can't easily exchange information between processes. You'll have to use some form of IPC. Android usually uses intents/broadcasts for that, or content providers. Apart from that, files can be used sometimes, but I think that would be overkill in your case...

**Comment by Jasi2169** [comment]
well that makes some sense,probably i gotta use intent ,never used content providers before
